packages/ra-core/src/controller/list/useList.storeKey.stories.tsx

```diff
@@ -0,0 +1,154 @@
+    sort,
+}: {
+    storeKey: string | false;
+    sort?: SortPayload;
+}) => {
+    const params = useList({
+        resource: 'posts',
+        perPage: 3,
+        sort,
+        storeKey,
+    });
+
+    return (
+        <div>
+            <span aria-label="storeKey" data-value={storeKey}>
+                storeKey: {storeKey}
+            </span>
+            <br />
+            <span aria-label="perPage" data-value={params.perPage}>
+                perPage: {params.perPage}
+            </span>
+            <br />
+            <button
+                aria-label="incrementPerPage"
+                onClick={() => {
+                    return params.setPerPage(++params.perPage);
+                }}
+            >
+                Increment perPage
+            </button>
+            <ul style={styles.ul}>
+                {params.data?.map(post => (
+                    <li key={post.id}>
+                        {post.title} - {post.votes} votes
+                    </li>
+                ))}
+            </ul>
+        </div>
+    );
+};
+
+const Layout = (props: CoreLayoutProps) => (
+    <div style={styles.mainContainer}>
+        <Link aria-label="top" to={`/top`}>
+            Go to Top Posts
+        </Link>
+        <Link aria-label="flop" to={`/flop`}>
+            Go to Flop Posts
+        </Link>
+        <Link aria-label="store" to={`/store`}>
+            Go to Store List
+        </Link>
+        <Link aria-label="nostore" to={`/nostore`}>
+            Go to No-Store List
+        </Link>
+
+        <br />
+        {props.children}
+    </div>
+);
+
+const TopPosts = (
+    <OrderedPostList
+        storeKey="topPostsKey"
+        sort={{ field: 'votes', order: 'DESC' }}
+    />
+);
+const FlopPosts = (
+    <OrderedPostList
+        storeKey="flopPostsKey"
+        sort={{ field: 'votes', order: 'ASC' }}
+    />
+);
+const StorePosts = (
+    <OrderedPostList
+        storeKey="storeListKey"
+        sort={{ field: 'votes', order: 'ASC' }}
+    />
+);
+const NoStorePosts = (
+    <OrderedPostList storeKey={false} sort={{ field: 'votes', order: 'ASC' }} />
+);
+
+export const ListsWithStoreKeys = () => (
+    <CoreAdminContext store={localStorageStore()} dataProvider={dataProvider}>
+        <CoreAdminUI layout={Layout}>
+            <CustomRoutes>
+                <Route path="/top" element={TopPosts} />
+                <Route path="/flop" element={FlopPosts} />
+            </CustomRoutes>
+            <Resource name="posts" />
+        </CoreAdminUI>
+    </CoreAdminContext>
+);
+
+export const ListsWithoutStoreKeys = () => (
+    <CoreAdminContext store={localStorageStore()} dataProvider={dataProvider}>
+        <CoreAdminUI layout={Layout}>
+            <CustomRoutes>
+                <Route path="/store" element={StorePosts} />
+                <Route path="/nostore" element={NoStorePosts} />
+            </CustomRoutes>
+            <Resource name="posts" />
+        </CoreAdminUI>
+    </CoreAdminContext>
+);
```

Comment:
This Story is a bit misleading, `StorePosts` is still using a `storeKey`.

Besides, `storeKey={false}` is not the same as having no `storeKey` (for `NoStorePosts` ). I'd expect both lists in `ListsWithoutStoreKeys` to have no `storeKey`.